src/pages/s/[classID]/teachers.tsx

```diff
@@ -0,0 +1,362 @@
+      Title={
+        <Title
+          name={{ title: t("title") }}
+          pageIcon={<MaterialIcon icon="school" />}
+          backGoesTo="/account/login"
+          LinkElement={Link}
+        />
+      }
+    >
+      <ListSection>
+        <CardList
+          listGroups={teacherList}
+          ListItem={({ content, className, onClick, id }) => {
+            return (
+              <button
+                onClick={() => {
+                  onClick();
+                  setShowMain(content);
+                }}
+                className="!w-full"
+              >
+                <TeacherCard
+                  key={content.id}
+                  teacher={content}
+                  hasSubjectSubgroup
+                  className={className}
+                />
+              </button>
+            );
+          }}
+          onChange={() => {}}
+        />
+      </ListSection>
+      <MainSection>
+        <Section className="!flex !flex-col !gap-4 !font-display">
+          <Section>
+            <div className="!sm:gap-6 !md:grid-cols-[1fr_5fr] grid !grid-cols-[1fr_3fr] items-stretch !gap-4">
+              <div className="aspect-square overflow-hidden rounded-xl sm:rounded-2xl">
+                <ProfilePicture src={showMain.profile} />
+              </div>
+              <div className="!flex !flex-col !justify-between">
+                <div>
+                  <h2 className="text-4xl font-bold">
+                    {nameJoiner(locale, showMain.name)}
+                  </h2>
+                  <p className="text-2xl font-medium">
+                    {showMain.subjectsInCharge[0].subjectSubgroup.name[locale]}
+                  </p>
+                </div>
+                <div>
+                  {showMain.classAdvisorAt ? (
+                    <p className="text-2xl font-medium">
+                      {t("advisor", {
+                        className: showMain.classAdvisorAt.name[locale],
+                      })}
+                    </p>
+                  ) : (
+                    <></>
+                  )}
+                </div>
+              </div>
+            </div>
+          </Section>
+          <Section>
+            <h3 className="text-3xl font-bold">{t("contacts")}</h3>
+            <ul className="layout-grid-cols-2">
+              {showMain.contacts.map((contact) => (
+                <ContactChip
+                  key={contact.id}
+                  contact={contact}
+                  className="!w-initial"
+                />
+              ))}
+            </ul>
+          </Section>
+          <Section>
+            <h3 className="text-3xl font-bold">{t("subjects")}</h3>
+            <ul className="!flex flex-col gap-2">
+              {showMain.subjectsInCharge.map((subjects) => (
+                <li>
+                  <Card type="horizontal">
+                    <CardHeader
+                      icon={
+                        <MaterialIcon
+                          icon={
```

Comment:
This will be changed to JSON map since `subjectSubgroup` is removed in the latest commit